R/util-ExtractSnapshotDate.R

```diff
@@ -0,0 +1,35 @@
+#' Extract the snapshot dates of snapshot directories.
+#'
+#' @param snapshot_paths `vector` a vector of Paths that contain snapshots, output of `ExtractDirectoryPaths`.
+#'
+#' @return `data.frame` a data.frame containing the foldername and the associated snapshot date.
+#'
+#' @importFrom tibble rownames_to_column
+#'
+#' @keywords internal
+#'
+#' @export
+ExtractSnapshotDate <- function(snapshot_paths){
+  acceptable_files <- c("meta_param.csv",
+                        "meta_workflow.csv",
+                        "results_analysis.csv",
+                        "results_bounds.csv",
+                        "results_summary.csv",
+                        "status_param.csv",
+                        "status_site.csv",
+                        "status_study.csv",
+                        "status_workflow.csv")
+
+  output <- purrr::map(snapshot_paths, ~paste0(., "/", acceptable_files)) %>%
+    purrr::set_names(stringr::str_extract(snapshot_paths, "([^/]+$)")) %>%
+    map(., ~read.csv(.[file.exists(.)][1], nrows=1) %>%
+          pull(gsm_analysis_date)) %>%
+    tibble::as_tibble() %>%
+    t() %>%
+    as.data.frame() %>%
+    tibble::rownames_to_column() %>%
+    `colnames<-`(c("foldername", "snapshot_date")) %>%
+    mutate(snapshot_date = as.Date(snapshot_date))
```

Comment:
This is great! I wanted to understand what was going on here, and am showing a different idea since `imap` could be helpful for slightly shorter code. Feel free to change or keep as-is.

```
  output <- purrr::map(snapshot_paths, ~paste0(., "/", acceptable_files)) %>%
    purrr::set_names(stringr::str_extract(snapshot_paths, "([^/]+$)")) %>%
    purrr::imap(., ~read.csv(.[file.exists(.)][1], nrows=1) %>% 
                  mutate(foldername = as.Date(.y))) %>% 
    bind_rows() %>% 
    select("foldername", "snapshot_date" = "gsm_analysis_date")
```

Reply:
I like it thanks!